pandas/io/formats/style.py

```diff
@@ -159,7 +164,8 @@ def __init__(
         self.index = data.index
         self.columns = data.columns
 
-        self.uuid = uuid
+        self.uuid_len = uuid_len if uuid_len < 33 else 32
+        self.uuid = (uuid or uuid4().hex[: self.uuid_len]) + "_"
```

Comment:
no replacing - with _ ? why is that

Reply:
hex representation of uuid is 32 hex char string. no "-" are included, so no replace needed.

underscore at the end is to improve HTML readability.